source/s3_auto_ranged_put.c

```diff
@@ -174,6 +145,28 @@ static void s_s3_meta_request_auto_ranged_put_destroy(struct aws_s3_meta_request
     aws_mem_release(meta_request->allocator, auto_ranged_put);
 }
 
+static void s_s3_auto_ranged_put_finish(
+    struct aws_s3_meta_request *meta_request,
+    struct aws_s3_request *failed_request,
+    int status_code,
+    int error_code) {
+    (void)status_code;
+    struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
+    aws_s3_meta_request_lock_synced_data(meta_request);
+    bool active = meta_request->synced_data.state == AWS_S3_META_REQUEST_STATE_ACTIVE;
+    if (active) {
+        meta_request->synced_data.state = AWS_S3_META_REQUEST_STATE_CANCELLING;
+    }
+    aws_s3_meta_request_unlock_synced_data(meta_request);
+    s_s3_auto_ranged_put_lock_synced_data(auto_ranged_put);
+    auto_ranged_put->synced_data.error_code = error_code;
+    auto_ranged_put->synced_data.failed_request = failed_request;
```

Comment:
These are currently ref counted right now.  (Arguably shouldn't be, and will be trying to take that out eventually.)  Need an aws_s3_request_acquire here, and then a release when you get rid of it.

Reply:
Looking at this with more awake eyes, you do need an acquire here.  And then you should be able to release it both in the destroy (to be safe) and after anywhere you call aws_s3_meta_request_finish_default when passing it in.

Reply:
yeah, I feel we do need a lock as well.😀